packages/services/ai-vercel/src/message-converter.ts

```
@@ -25,16 +29,56 @@ function convertToSDKMessage(msg: AIMessage): ModelMessage {
+      return {
+        role: 'system',
+        content:
+          typeof msg.content === 'string'
+            ? msg.content
+            : ((msg.content ?? '') as string),
+      }
     case 'user':
-      return { role: 'user', content: msg.content ?? '' }
-    case 'assistant':
+      if (Array.isArray(msg.content)) {
+        const parts = (msg.content as AIContentPart[]).map((part) => {
```

Biome (2.4.4)

[error] This callback passed to map() iterable method should always return a value.

(lint/suspicious/useIterableCallbackReturn)